### PR TITLE
Update intravisit docs

### PR DESCRIPTION
**Pull request by camsteffen**
Follow-up to #90986.

r? @cjgillot

### CODE REVIEW

File: compiler/rustc_hir/src/intravisit.rs [hunk 1]
```diff
@@ -217,42 +222,38 @@ use nested_filter::NestedFilter;
...
     }
 
-    /// Invoked when a nested item is encountered. By default does
-    /// nothing unless you override `nested_visit_map` to return other than
-    /// `None`, in which case it will walk the item. **You probably
-    /// don't want to override this method** -- instead, override
-    /// `nested_visit_map` or use the "shallow" or "deep" visit
+    /// Invoked when a nested item is encountered. By default does nothing
+    /// unless you override `Self::NestedFilter` to a type other than
+    /// `nested_filter::None`, in which case it will walk the item. **You
```
Comment (cjgillot) [inline compiler/rustc_hir/src/intravisit.rs:251]: I'd say: "By default, when `Self::NestedFilter` is `nested_filter::None`, this method does nothing.

File: compiler/rustc_hir/src/intravisit.rs [hunk 2]
```diff
@@ -291,9 +292,9 @@ pub trait Visitor<'v>: Sized {
     }
 
     /// Invoked to visit the body of a function, method or closure. Like
-    /// visit_nested_item, does nothing by default unless you override
-    /// `nested_visit_map` to return other than `None`, in which case it will walk
-    /// the body.
+    /// `visit_nested_item`, does nothing by default unless you override
+    /// `Self::NestedFilter` to a type other than `None`, in which case
```
Comment (cjgillot) [inline compiler/rustc_hir/src/intravisit.rs:296]:
```suggestion
    /// `Self::NestedFilter`.
```
and stop the doc here. The description on `visit_nested_item`should be enough.

File: compiler/rustc_hir/src/itemlikevisit.rs [hunk 3]
```diff
@@ -17,8 +17,8 @@ use super::{ForeignItem, ImplItem, Item, TraitItem};
 ///    an item, but don't care about how item-like things are nested
 ///    within one another.
 ///    - Example: Examine each expression to look for its type and do some check or other.
-///    - How: Implement `intravisit::Visitor` and override the `nested_visit_map()` method
-///      to return `NestedVisitorMap::OnlyBodies` and use
+///    - How: Implement `intravisit::Visitor` and override the `NestedFilter` type to
+///      `nested_filter::OnlyBodies` (and implement `nested_visit_map`), and use
 ///      `tcx.hir().krate().visit_all_item_likes(&mut visitor.as_deep_visitor())`. Within
```
Comment (cjgillot) [inline compiler/rustc_hir/src/itemlikevisit.rs:22]:
```suggestion
///      `tcx.hir().visit_all_item_likes(&mut visitor.as_deep_visitor())`. Within
```

**Review by cjgillot — COMMENTED** [review]
Thanks @camsteffen! Just a few nits.

**Comment by camsteffen** [comment]
Addressed nits. Will squash if all is well.

File: compiler/rustc_hir/src/itemlikevisit.rs [hunk 4]
```diff
@@ -29,9 +29,9 @@ use super::{ForeignItem, ImplItem, Item, TraitItem};
 ///    item-like things.
 ///    - Example: Lifetime resolution, which wants to bring lifetimes declared on the
 ///      impl into scope while visiting the impl-items, and then back out again.
-///    - How: Implement `intravisit::Visitor` and override the `nested_visit_map()` method
-///      to return `NestedVisitorMap::All`. Walk your crate with `intravisit::walk_crate()`
-///      invoked on `tcx.hir().krate()`.
+///    - How: Implement `intravisit::Visitor` and override the `NestedFilter` type to
+///      `nested_filter::All` (and implement `nested_visit_map`). Walk your crate with
+///      `intravisit::walk_crate()` invoked on `tcx.hir().krate()`.
```
Comment (cjgillot) [inline compiler/rustc_hir/src/itemlikevisit.rs:34]: `walk_crate` does not exist any more. The user should call `tcx.hir().walk_toplevel_module(visitor)`.

**Comment by cjgillot** [comment]
Thanks! r=me with one nit + squashed.

**Comment by JohnCSimon** [comment]
@camsteffen 

Ping from triage: Can you please post the status of this PR?

**Comment by camsteffen** [comment]
This should be ready now but, since some time has passed, I'll let @cjgillot review once more in case anything has changed.

**Pull request merged**